storage/txcache/maps/bucketSortedMap.go

```diff
@@ -78,6 +78,13 @@ func (sortedMap *BucketSortedMap) NotifyScoreChangeByKey(key string) {
 
 // NotifyScoreChange moves or adds the item to the corresponding score chunk
 func (sortedMap *BucketSortedMap) NotifyScoreChange(item BucketSortedMapItem) {
+	flag := item.GetScoreChangeInProgressFlag()
+	if flag.IsSet() {
```

Comment:
this does not work as expected. Concurrency issues still remain

Reply:
Good finding, fixed.